ACHNBrowserUI/ACHNBrowserUI/views/todayDashboard/TodayTasksSection.swift

```diff
@@ -48,32 +55,36 @@ struct TodayTasksSection: View {
             if !task.hasProgress {
                 return
             }
-            self.collection.updateProgress(taskName: taskName)
+            self.collection.updateProgress(taskId: taskId)
             FeedbackGenerator.shared.triggerSelection()
         }
         .onLongPressGesture {
             guard task.hasProgress else { return }
-            self.collection.resetProgress(taskName: taskName)
+            self.collection.resetProgress(taskId: taskId)
             FeedbackGenerator.shared.triggerSelection()
         }
     }
     
     var body: some View {
         Section(header: SectionHeaderView(text: "Today's Tasks", icon: "checkmark.seal.fill")) {
             VStack(spacing: 15) {
-                HStack {
-                    makeTaskBubble(icon: "icon-iron", taskName: DailyTasks.taskName.rocks)
-                    makeTaskBubble(icon: "icon-villager", taskName: DailyTasks.taskName.villagers)
-                    makeTaskBubble(icon: "icon-fossil", taskName: DailyTasks.taskName.fossils)
-                    makeTaskBubble(icon: "icon-leaf", taskName: DailyTasks.taskName.furniture)
-                }
-                HStack {
-                    makeTaskBubble(icon: "icon-bell", taskName: DailyTasks.taskName.bell)
-                    makeTaskBubble(icon: "icon-miles", taskName: DailyTasks.taskName.nookmiles)
-                    makeTaskBubble(icon: "icon-bottle-message", taskName: DailyTasks.taskName.bottle)
-                    makeTaskBubble(icon: "icon-recipe", taskName: DailyTasks.taskName.villagerHouses)
+                ForEach(0 ..< self.rows) { row in
+                    HStack {
+                        ForEach(row*4 ..< min(row*4+4, self.tasksCount)) { index in
+                            self.makeTaskBubble(taskId: index)
+                        }
+                    }
                 }
                 HStack {
```

Comment:
Just noticed here that `Edit` and `Reset` are very close to each others
![image](https://user-images.githubusercontent.com/2297556/83348088-76c39980-a321-11ea-9c75-8ab0ae915511.png)
if you just add three Spacers you will have (IMO) a better UI:
```swift
HStack {
    Spacer()
    Text("Edit")
        ...
    Spacer()
    Text("Reset")
        ...
    Spacer()
}
```
![image](https://user-images.githubusercontent.com/2297556/83348174-f2bde180-a321-11ea-8173-70a5e1ba4e8f.png)